src/sentry/static/sentry/app/views/stream.jsx

```diff
@@ -116,19 +111,25 @@ const Stream = createReactClass({
     this.fetchSavedSearches();
     this.fetchProcessingIssues();
     this.fetchTags();
-
-    // Make sure it gets called on mount
-    this.onLatestContextChange(LatestContextStore.getInitialState());
   },
 
   componentWillReceiveProps(nextProps) {
     if (this.state.loading) {
       return;
     }
 
-    // Do not make new API request if props haven't actually changed
-    // Unless no request has been performed yet
-    if (!_.isEqual(this.props, nextProps) || !this.lastRequest) {
```

Comment:
What is `lastRequest` here for? @MaxBittker? Can it be cleaned up?

Reply:
I've added it recently [here](https://github.com/getsentry/sentry/commit/0c98030d17a07f17ccf81f35457b089df8681389#diff-6c99060bedfae86add32892f76673815) but I realised it's not quite correct. Since it gets set to null after the request is done, this expression essentially always gets evaluated to true. This condition was intended to improve performance by potentially skipping an extra API call but it doesn't quite work.